sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -1009,18 +1009,19 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]]
     append(str)
     append("\n")
 
-    if (innerChildren.nonEmpty) {
+    val innerChildrenLocal = innerChildren
```

Comment:
why do we need a new local variable?

Reply:
it is possbile innerChildren is not a variable, it is defined as `def innerChildren`

Reply:
`children` is `def` too, but I don't see we create local variables...

Anyway, this is safer, I'm fine with it